drivers/input/input_chsc5x.c

```diff
@@ -0,0 +1,196 @@
+};
+
+#define CHSC5X_BASE_ADDRESS       0x20, 0x00, 0x00
+#define CHSC5X_ADDRESS_IC_TYPE    0x81
+#define CHSC5X_ADDRESS_TOUCH_DATA 0x2C
+#define CHSC5X_SIZE_TOUCH_DATA    7
+
+#define DHSC5X_OFFSET_EVENT_TYPE    0x00
+#define DHSC5X_OFFSET_FINGER_NUMBER 0x01
+#define DHSC5X_OFFSET_X_COORDINATE  0x02
+#define DHSC5X_OFFSET_Y_COORDINATE  0x03
+#define DHSC5X_OFFSET_PRESSURE      0x04
+#define DHSC5X_OFFSET_XY_COORDINATE 0x05
+#define DHSC5X_OFFSET_TOUCH_EVENT   0x06
+
+LOG_MODULE_REGISTER(chsc5x, CONFIG_INPUT_LOG_LEVEL);
+
+static void chsc5x_work_handler(struct k_work *work)
+{
+	struct chsc5x_data *data = CONTAINER_OF(work, struct chsc5x_data, work);
+	const struct device *dev = data->dev;
+	const struct chsc5x_config *cfg = dev->config;
+	uint32_t row, col;
+	bool is_pressed;
+	int ret;
+	uint8_t read_buffer[CHSC5X_SIZE_TOUCH_DATA] = {0};
+	uint8_t write_buffer[4] = {CHSC5X_BASE_ADDRESS, CHSC5X_ADDRESS_TOUCH_DATA};
+
+	ret = i2c_write_read_dt(&cfg->i2c, write_buffer, sizeof(write_buffer), read_buffer,
+				sizeof(read_buffer));
+	if (ret < 0) {
+		LOG_ERR("Could not read data: %i", ret);
+		return;
+	}
+
+	is_pressed = !(read_buffer[DHSC5X_OFFSET_TOUCH_EVENT] & 0x40);
+	col = read_buffer[DHSC5X_OFFSET_X_COORDINATE] +
+	      ((read_buffer[DHSC5X_OFFSET_XY_COORDINATE] & 0x0f) << 8);
+	row = read_buffer[DHSC5X_OFFSET_Y_COORDINATE] +
+	      ((read_buffer[DHSC5X_OFFSET_XY_COORDINATE] & 0xf0) << 4);
+
+	if (is_pressed) {
+		input_touchscreen_report_pos(dev, col, row, K_FOREVER);
+	}
+
+	input_report_key(dev, INPUT_BTN_TOUCH, is_pressed, true, K_FOREVER);
+}
+
+static void chsc5x_isr_handler(const struct device *dev, struct gpio_callback *cb, uint32_t mask)
+{
+	struct chsc5x_data *data = CONTAINER_OF(cb, struct chsc5x_data, int_gpio_cb);
+
+	k_work_submit(&data->work);
+}
+
+static int chsc5x_chip_init(const struct device *dev)
+{
+	const struct chsc5x_config *cfg = dev->config;
+	uint8_t ic_type = 0;
+	int ret;
+	uint8_t write_buffer[4] = {CHSC5X_BASE_ADDRESS, CHSC5X_ADDRESS_IC_TYPE};
+
+	if (!i2c_is_ready_dt(&cfg->i2c)) {
+		LOG_ERR("I2C bus %s not ready", cfg->i2c.bus->name);
+		return -ENODEV;
+	}
+
+	ret = i2c_write_read_dt(&cfg->i2c, write_buffer, sizeof(write_buffer), &ic_type, 1);
+	if (ret < 0) {
+		LOG_ERR("Could not read data: %i", ret);
+		return ret;
+	}
+
+	switch (ic_type) {
+	case CHSC5X_IC_TYPE_CHSC5472:
+	case CHSC5X_IC_TYPE_CHSC5448:
+	case CHSC5X_IC_TYPE_CHSC5448A:
+	case CHSC5X_IC_TYPE_CHSC5460:
+	case CHSC5X_IC_TYPE_CHSC5468:
+	case CHSC5X_IC_TYPE_CHSC5432:
+	case CHSC5X_IC_TYPE_CHSC5816:
+	case CHSC5X_IC_TYPE_CHSC1716:
+		break;
+	default:
+		LOG_ERR("CHSC5X wrong ic type: returned 0x%x", ic_type);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int chsc5x_init(const struct device *dev)
+{
+	const struct chsc5x_config *config = dev->config;
+	struct chsc5x_data *data = dev->data;
+	int ret;
+
+	data->dev = dev;
+
+	k_work_init(&data->work, chsc5x_work_handler);
+
+	if (gpio_is_ready_dt(&config->reset_gpio)) {
```

Comment:
uh, is this a shortcut for not having an explicit check for `config->reset_gpio.port != NULL`? it's very unlikely it'll go wrong but I'd rather this check for port != null and have an explicit `gpio_is_ready_dt` error path, if anything for coherency with the other drivers